chore: clean up taxonomy inconsistencies (#2399)
mendersoftware/mender-docs: 04.Operating-System-updates-Debian-family/03.Customize-Mender/docs.md

## Mender Server address

> If you wish to use a self-hosted Mender Server instead of hosted Mender, you will need to provide a

Reviewer: Is it just me or does this sentence sound a bit strange? How about something like this?
```suggestion
If you wish to use a self-hosted Mender Server instead of the managed [Mender](https://hosted.mender.io) service, you will need to provide a
```

Author: @alfrunes I think you're right... I noticed we use all of 'on-premise', 'self-hosted', and 'self-managed' so I think first I should find out which one we want to use consistently going forward.

For now, maybe "self-managed installation" is currently used most throughout the docs, so how about:

```suggestion
For self-managed installations of the Mender server (not [hosted Mender](https://hosted.mender.io?target=_blank), you will need to provide a
```

Reviewer: Yup sounds good to me. As long as it doesn't read "... hosted instead of hosted..."
